### PR TITLE
PSP-9268 Add sub-interest files to a main file

### PR DESCRIPTION
**Pull request by asanchezr**
https://github.com/user-attachments/assets/0aff4c54-fb6e-48ab-b4d9-ba61f1a49c13

### CODE REVIEW

File: source/backend/dal/Repositories/AcquisitionFileRepository.cs [hunk 1]
```diff
@@ -798,16 +832,35 @@ public List<PimsAcquisitionFile> GetAcquisitionSubFiles(long acquisitionFileId,
...
         {
             return $"{prefix:00}-{fileNumber}-{suffix:00}";
         }
 
+        /// <summary>
+        /// Returns the suffix portion of the supplied Acquisition File Number.
+        /// </summary>
+        /// <param name="acquisitionFileNumber">The Acquisition File Number.</param>
+        /// <returns>The file number suffix (e.g. "1", "2", "3", etc) if it is found, or -1 if it is not.</returns>
+        private static int GetAcquisitionNumberSuffix(string acquisitionFileNumber)
```
Comment (eddherrera) [inline source/backend/dal/Repositories/AcquisitionFileRepository.cs:850]: Wouldn't it be better to make this an Extension Method of the PimsAcquisitionFile?

Comment (asanchezr) [reply source/backend/dal/Repositories/AcquisitionFileRepository.cs:850]: yeah - can do that so the logic can be reused in the future

File: source/backend/dal/Repositories/AcquisitionFileRepository.cs [hunk 2]
```diff
@@ -671,9 +671,23 @@ public PimsAcquisitionFile Add(PimsAcquisitionFile acquisitionFile)
...
+                // generate file number for "main" files
+                int nextFileNo = GetNextAcquisitionFileNumberSequenceValue();
+                acquisitionFile.FileNo = nextFileNo;
+                acquisitionFile.FileNumber = GenerateAcquisitionNumber(acquisitionFile.RegionCode, nextFileNo, 1);
+            }
+            else
+            {
+                // generate file number for "sub-files"
+                var parentFile = Context.PimsAcquisitionFiles
+                    .FirstOrDefault(x => x.AcquisitionFileId == acquisitionFile.PrntAcquisitionFileId) ?? throw new KeyNotFoundException();
```
Comment (eddherrera) [inline source/backend/dal/Repositories/AcquisitionFileRepository.cs:685]: .AsNoTracking()

Comment (asanchezr) [reply source/backend/dal/Repositories/AcquisitionFileRepository.cs:685]: Updated

File: source/backend/dal/Repositories/AcquisitionFileRepository.cs (hunk 1 above)
Comment (asanchezr) [reply source/backend/dal/Repositories/AcquisitionFileRepository.cs]: Updated

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**